facebook-commerce-events-tracker.php

```diff
@@ -187,6 +193,128 @@ function( $item ) {
 			$this->pixel->inject_event( $event_name, $event_data, 'trackCustom' );
 		}
 
+
+		/**
+		 * Attempts to add a session variable to indicate that a product search event occurred.
+		 *
+		 * The session variable is used to catch search events that have a single result.
+		 * In those cases WooCommerce redirects customers to the product page instead of showing the search results.
+		 *
+		 * The plugin can't inject a Pixel event code on redirect responses, but it can check for the presence of the variable on the product page.
+		 *
+		 * This method is hooked to woocommerce_redirect_single_search_result which is triggered right before redirecting.
+		 *
+		 * @internal
+		 *
+		 * @since 2.0.6-dev.1
+		 *
+		 * @param bool $redirect whether to redirect to the product page
+		 * @return bool
+		 */
+		public function maybe_add_product_search_event_to_session( $redirect ) {
+
+			if ( $redirect && $this->search_event && $this->is_single_search_result() ) {
+
+				$this->add_product_search_event_to_session( $this->search_event );
+			}
+
+			return $redirect;
+		}
+
+
+		/**
+		 * Determines whether the current request is a product search with a single result.
+		 *
+		 * @since 2.0.6-dev.1
+		 *
+		 * @return bool
+		 */
+		private function is_single_search_result() {
+			global $wp_query;
+
+			return is_search() && is_post_type_archive( 'product' ) && 1 === absint( $wp_query->found_posts );
```

Comment:
I believe if we re-ordered the conditions check would perform better and use less memory
```php
return is_search() && 1 === absint( $wp_query->found_posts ) && is_post_type_archive( 'product' );
```
That way it bail as long as the search result has a single post before checking if it's for a product post type or not. what do you think?

Reply:
That makes sense. Thanks! Fixed in b4c05624